src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int SIZE_OF_ARRAY = 6;
+
+    public static void main(String[] args) {
+        Figure[] figures = new Figure[SIZE_OF_ARRAY];
+        for (int i = 0; i < SIZE_OF_ARRAY; i++) {
+            if (i < 3) {
+                figures[i] = new FigureSupplier().getRandomFigure();
```

Comment:
Create an instance of FigureSupplier class outside the loop to avoid unnecessary object creation. 🛠️